controller/lib/hal/uart_dma.cpp

```diff
@@ -212,50 +225,62 @@ void UART_DMA::UART_ISR() {
     uart->intClear.s.rtocf = 1;
 
     // TODO define logic if stopRX() has to be here
-    rxListener.onRxError(e);
+    if (rxListener) {
+      rxListener->onRxError(e);
+    }
   }
 
   if (isCharacterMatchInterrupt()) {
     uart->request.s.rxfrq = 1; // Clear RXNE flag before clearing other flags
     uart->intClear.s.cmcf = 1; // Clear char match flag
     // TODO define logic if stopRX() has to be here
-    rxListener.onCharacterMatch();
+    if (rxListener) {
+      rxListener->onCharacterMatch();
+    }
```

Comment:
Please add some comments to the beginning of this method why it handles only the error and character match, but not RxComplete.